server/src/main/java/org/apache/iotdb/db/conf/IoTDBConfigCheck.java

```diff
@@ -84,12 +84,14 @@ private void createDir(String filepath) {
   }
 
   private void checkFile(String filepath) {
-    // create file : read timestamp precision from engine.properties, create system_properties.txt
+    // create file : read timestamp precision from engine.properties, create system.properties
     // use output stream to write timestamp precision to file.
     File file = SystemFileFactory.INSTANCE
             .getFile(filepath + File.separator + PROPERTIES_FILE_NAME);
+    File tmpPropertiesFile = new File(file.getAbsoluteFile() 
+        + File.separator + "tmp");
     try {
-      if (!file.exists()) {
+      if (!file.exists() && !tmpPropertiesFile.exists()) {
```

Comment:
when the `file` exists during upgrade, both the if and elif does not work

Reply:
Yes, you are right. When the file exists, both the if and elif does not work. But following this part, I will check if the file need to be upgraded. I don't think there is any problem.